pkg/middleware/stored_session_test.go

```diff
@@ -17,9 +18,104 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+type TestLock struct {
+	Locked            bool
+	WasObtained       bool
+	WasRefreshed      bool
+	WasReleased       bool
+	PeekedCount       int
+	LockedOnPeekCount int
+	ObtainError       error
+	PeekError         error
+	RefreshError      error
+	ReleaseError      error
+}
+
+func (l *TestLock) Obtain(_ context.Context, _ time.Duration) error {
+	if l.ObtainError != nil {
+		return l.ObtainError
+	}
+	l.Locked = true
+	l.WasObtained = true
+	return nil
+}
+
+func (l *TestLock) Peek(_ context.Context) (bool, error) {
+	if l.PeekError != nil {
+		return false, l.PeekError
+	}
+	locked := l.Locked
+	l.Locked = false
```

Comment:
I was having a look at this again and noticed this line, this seems like the peek implementation of this test lock is creating side effects, I wouldn't expect Peeking to cause an unlock.

If we comment out this line, the test `when the session is locked and instead loaded from storage` hangs forever

Reply:
I think the only way to solve this, is to use an expire time for the lock. But this would mean I can't predict the number of peeks anymore.